File: Codeforces/codeTon3/B.py

def inp():
    return list(map(int, input().split()))

def convolve(a):
    ans = []
    if len(a) == 0:
        return ans
    c = a[0]
    n = 0
    for i in a:
        if i == c:
            n += 1
        else:
            ans.append((c, n))
            c = i
            n = 1
    ans.append((c, n))
    return ans

for _ in range(int(input())):
    n = int(input())
    a = input()
    ans = 0
    cnv = convolve(a)
    for i in cnv:
        ans = max(ans, i[1] * i[1])
    print(max(ans, a.count('0') * a.count('1')))

File: Codeforces/codeTon3/B.cpp

#include <iostream>
#include <string>
#include <vector>

using namespace::std;
#define int long long

signed main()
{
    int t;
    cin >> t;
    while (t--) {
        int _n;
        string s;
        cin >> _n >> s;
        int _0 = 0, _1 = 0;
        int n = 0, ans = 0;
        char cc = s.at(0);
        for (char c : s) {
            if (c == '1')
                ++_1;
            else
                ++_0;
            if (cc == c)
                ++n;
            else {
                ans = max(n * n, ans);
                cc = c;
                n = 1;
            }
        }
        ans = max(n * n, ans);
        ans = max(ans, _1 * _0);
        cout << ans << endl;
    }
}
